server/src/cm_job_task.cpp

```diff
@@ -15675,6 +15690,173 @@ ts_generate_cert (nvplist * req, nvplist * res, char *_dbmt_error)
     return ret_val;
 }
 
+int
+ts_start_statdump (nvplist *req, nvplist *res, char *_dbmt_error)
+{
+  int ret_val = ERR_NO_ERROR;
+  char *db_name, *interval_str;
+  int interval = 0;
+  char *argv [10];
+  char path [512];
+  int argc = 0;
+  char note [20];
+  int slot = -1;
+  db_name = nv_get_val (req, "_DBNAME");
+  interval_str = nv_get_val (req, "interval");
+  if (!interval_str || !db_name)
+   {
+     nv_update_val (res, "note", "no sufficient arguments");
+     LOG_ERROR ("start_statdump: dbname or interval was not specified");
+     return -1;
+   }
+
+  if (find_statdumpd_info (db_name) >= 0)
+    {
+      nv_update_val (res, "note", "already running");
+      return -1;
+    }
+
+  slot = find_new_statdumpd_info ();
+  if (slot < 0)
+    {
+      nv_update_val (res, "note", "memory allocation error");
```

Comment:
If slot is full, I think you'd better to check this error and print another message such as "max slot error".

Reply:
got it.